logic/vote.js

```diff
@@ -95,6 +96,10 @@ Vote.prototype.verify = function (trs, sender, cb) {
 		return setImmediate(cb, 'Voting limit exceeded. Maximum is 33 votes per transaction');
 	}
 
+	if (trs.asset.votes.length > _.uniq(trs.asset.votes).length) {
```

Comment:
We should compare public keys instead of votes, that will cover unvote/downvote in same transaction.